api/tests/integration/infrastructure/utils/bookshelf-to-domain-converter_test.js

```diff
@@ -58,23 +56,11 @@ describe('Integration | Infrastructure | Utils | Bookshelf to domain converter',
       expect(domainObject.scorecards).to.deep.equal([]);
 
     });
-    it('should support has-one relationships', async () => {
-      // given
-      const userId = databaseBuilder.factory.buildUser().id;
-      const campaignParticipation = databaseBuilder.factory.buildCampaignParticipation({ userId });
-      const assessmentId = databaseBuilder.factory.buildAssessment({ userId, campaignParticipationId: campaignParticipation.id }).id;
-
-      await databaseBuilder.commit();
-      const bookshelfObject = await BookshelfAssessment.where({ id: assessmentId }).fetch({
-        withRelated: ['campaignParticipation'],
-      });
 
-      // when
-      const domainObject = bookshelfToDomainConverter.buildDomainObject(BookshelfAssessment, bookshelfObject);
-
-      // then
-      expect(domainObject.campaignParticipation).to.be.instanceOf(CampaignParticipation);
+    it('should support has-one relationships', async () => {
+      // TODO : Il n'y a pas d'exemple d'objet du Domain qui récupère un autre objet du Domain via hasOne.
```

Comment:
j'ai pas compris pourquoi on supprime ce test. Le fonctionnel n'est pas fonctionnel ?

Reply:
Si j'ai bien compris, notre converter ne peut plus s'appuyer sur un exemple de code basé sur `hasOne`. 

La relation entre un assessment et une campaignParticipation était la seule de notre code. 

En la modifiant, il n'y plus de cas concret pour valider le fonctionnement de notre convertisseur ; d'où la suppression du test qui ne peut plus appeler d'éléments de code pour validation

Reply:
et dans Certification-Course : `this.hasOne('Assessment', 'courseId')` ?

Reply:
Effectivement, bien vu ! 
J'ai fais quelques essais et Mel était probablement tombé sur le même cas. En fait, on peut faire le test mais l'objet du domaine`CertificationCourse` ne porte pas d'attribut `assessment` en `includes` ce qui fait qu'au moment où on a le `expect(domainObject.assessment).to.be.instanceOf(Assessment);`, on a un `undefined`.

Est-ce qu'on doit vraiment l'inclure ?

Reply:
Il faudrait plus creuser, mais je pense que c'est une incohérence entre le modèle  d'infrastructure et le modèle du domaine. La méthode `get` de `certificationCourseRepository` va bien chercher en même temps l'`assessment` et le convertit après en modèle du domaine avec le `_toDomain`, où il y a bien l'attribut `assessment` de précisé. Et après dans le domaine (à 3 endroits, on ne s'en sert pas, et on va chercher de nouveau l'assessment associé). Je pense qu'il y a une amélioration/refacto à étudier voire faire sur ce point.
Concrètement, dans ce ticket, peut-être juste rajouter le `includes`. Tu as essayé ? Ça a des impacts trop importants ou pas ?

Reply:
Le "cas certificationCourse" a commencé à faire l'objet d'une discussion de refacto important car il est plein de d'incohérences  (voir ici pour les curieux : https://github.com/1024pix/pix/issues/670).
Il reste en effet toujours la contrainte exprimée en bookshelf de certifCourse.hasOne(assessment) mais dans les faits, cette relation est fausse (il existe dans la base de données prod de moult exemples de certificationCourse qui ont plusieurs assessments.

Reply:
Je vous laisse le bébé alors :)